llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -142,6 +142,7 @@ def hasLDU : Predicate<"Subtarget->hasLDU()">;
 def hasPTXASUnreachableBug : Predicate<"Subtarget->hasPTXASUnreachableBug()">;
 def noPTXASUnreachableBug : Predicate<"!Subtarget->hasPTXASUnreachableBug()">;
 def hasOptEnabled : Predicate<"TM.getOptLevel() != CodeGenOptLevel::None">;
+def hasAAFeatures : Predicate<"Subtarget->hasAAFeatures()">;
```

Comment:
Nit: it's an unfortunate name choice, as `AA` commonly stands for Alias Analysis in LLVM.

Perhaps rename the predicate to `hasAcceleratedFeatures` to match the nomenclature used in PTX docs.

Reply:
Also, we already have `hasSM90a` predicate : https://github.com/llvm/llvm-project/blob/a57bbff736ff4658940266a218491ea9d576b13b/llvm/lib/Target/NVPTX/NVPTXInstrInfo.td#L175

Reply:
Yes, but it uses for all sm >= 90 with feature "a"